infra/gcp/terraform/k8s-infra-elections/database.tf

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+resource "google_sql_database" "database" {
+  name     = var.db_name
+  project  = google_project.project.project_id
+  instance = google_sql_database_instance.db_instance.name
+}
+
+resource "google_sql_database_instance" "db_instance" {
+  project             = google_project.project.project_id
+  database_version    = var.db_version
+  region              = "us-central1"
+  deletion_protection = true
+
+  settings {
+
+    tier                  = var.cloudsql_tier
+    disk_size             = var.cloudsql_disk_size_gb
+    disk_autoresize_limit = var.disk_autoresize_limit
+    disk_type             = var.disk_type
+    disk_autoresize       = var.disk_autoresize
+    availability_type     = "REGIONAL"
+
+    ip_configuration {
+      ipv4_enabled    = false
+      private_network = "projects/${google_project.project.project_id}/global/networks/default"
+    }
+
+    database_flags {
+      name  = "autovacuum"
+      value = "on"
+    }
+
+    database_flags {
+      name  = "max_connections"
+      value = var.cloudsql_max_connections
+    }
+
+    database_flags {
+      name  = "cloudsql.enable_pgaudit"
+      value = "on"
+    }
+
+    database_flags {
+      name  = "pgaudit.log"
+      value = "all"
+    }
+
+    backup_configuration {
+      enabled    = true
+      location   = var.cloudsql_backup_location
+      start_time = "02:13"
+    }
+
+    maintenance_window {
+      day          = 7
+      hour         = 2
+      update_track = "stable"
+    }
```

Comment:
Would it be wise to have backups happen sometime that doesn't overlap with (possibly precedes) the maintenance window?

Reply:
Moved the maintenance window to Friday, 04:00 UTC.